### ISSUE TITLE
QoL: "Salvage Item/All xx Item" context menu addition. (+2)

### ISSUE THREAD

**Issue by raben1** [gwdevhub/GWToolboxpp#1134]
- The same way Store is in the context menu could go one for salvage option. This way you can selectively salvage items without the risk of salvage items you don't want to (the only reason why I don't use the "Salvage all ..." of Salvage kits). Even a shortcut like shift+ctrl+click (dunno) could be amazing, similar to the store one.

- Also the feature to un-tick other context menu options like Store (ctrl+click) or Salvage all items.

- Another suggestion is to add the possibility to bypass the identification need of blue/purple items before salvage them, so you can salvage them directly without identify. (Or auto id if this can't be skipped).

**Comment by raben1** [comment]
![context menu custom](https://github.com/gwdevhub/GWToolboxpp/assets/123206408/2b3964c7-9381-43c8-bb68-d3da8c8e0e36)

**Comment by DubbleClick** [comment]
Point 1: That's way overkill for the rather irrelevant and niche functionality.

Point 2: A setting for this seems highly unnecessary as well. It will only lead to confusion among users when they untick a checkbox and then fail to find the option in the future.

Point 3: Yeah that's a nice idea. Rather surprised that the functionality right now only allows salvaging identified items, will have to check that before implementing.

**Comment by RiftLurker** [comment]
> Another suggestion is to add the possibility to bypass the identification need of blue/purple items before salvage them, so you can salvage them directly without identify. (Or auto id if this can't be skipped).

> Point 3: Yeah that's a nice idea. Rather surprised that the functionality right now only allows salvaging identified items, will have to check that before implementing.

This is a guild wars bug btw, not something the salvage function explicitly doesn't salvage.

**Comment by raben1** [comment]
> Point 1: That's way overkill for the rather irrelevant and niche functionality.

Despite I agree with you, isn't everything in this game niche, including the game itself? :D
That screenshot looks nice though, as much as the 25 slot Backpack there.

**Comment by 3vcloud** [comment]
Items need to be identified so you don’t get random upgrades in your inventory when you salvage an unidentified item. Would get very annoying when you go to salvage all and it throws a load of crap into your bag instead of crunching stuff down into mats

**Comment by raben1** [comment]
Yeah, of course. I meant to salvage with regular Salvage Kit, that only produces mats

**Comment by 3vcloud** [comment]
Ok so you can see why this isn’t going to work very well; it will have to only use regular salvage kits, otherwise it’ll only salvage identified items. It’s too much code debt just to confuse a user who just wants to clear out their inventory with a convenience feature
